test/longest_prefix_test.ml

```diff
@@ -0,0 +1,115 @@
+open Lib
+open Github_t
+open Slack_message
+
+let single_file =
+  [
+    {
+      sha = "something something";
+      filename = "testdir1/testdir2/changed_file.txt";
+      status = "something something";
+      additions = 5;
+      deletions = 5;
+      changes = 10;
+      url = "test_link";
+    };
+  ]
+
+let multiple_files_same_dir =
+  [
+    {
+      sha = "something something";
+      filename = "testdir1/testdir2/changed_file2.txt";
```

Comment:
if only filename is relevant, should have function to construct record given a filename and apply it to a list of filenames for more readable tests